source/fab/tasks/fortran.py

```diff
@@ -464,80 +486,110 @@ def run(self):
                         raise TaskException(
                             end_message.format(**end_values))
 
+        return [new_artifact]
+
+
+class FortranPreProcessor(Task):
+    def __init__(self,
+                 preprocessor: str,
+                 flags: List[str],
+                 workspace: Path):
+        self._preprocessor = preprocessor
+        self._flags = flags
+        self._workspace = workspace
+
+    def run(self, artifacts: List[Artifact]) -> List[Artifact]:
 
-class FortranPreProcessor(SingleFileCommand):
+        if len(artifacts) == 1:
+            artifact = artifacts[0]
+        else:
+            msg = ('Fortran Preprocessor expects only one Artifact, '
+                   f'but was given {len(artifacts)}')
+            raise TaskException(msg)
 
-    @property
-    def as_list(self) -> List[str]:
-        base_command = ['cpp', '-traditional-cpp', '-P']
-        file_args = [str(self._filename), str(self.output[0])]
-        return base_command + self._flags + file_args
+        artifact = artifacts[0]
```

Comment:
Hasn't this already been done in line 504?

Reply:
So it has!  I forgot to take it out after adding the trapping above; thanks :+1: